modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java

```diff
@@ -52,866 +54,960 @@
+                                DevicePolicyManager.PASSWORD_QUALITY_ALPHANUMERIC);
+                    }
+                } else if (policyData.get(resources.getString(
+                        R.string.policy_password_require_alphanumeric)) instanceof Boolean) {
+                    isAlphanumeric =
+                            policyData.getBoolean(resources.getString(
+                                    R.string.policy_password_require_alphanumeric));
+                    if (isAlphanumeric) {
+                        devicePolicyManager.setPasswordQuality(demoDeviceAdmin,
+                                DevicePolicyManager.PASSWORD_QUALITY_ALPHANUMERIC);
+                    }
+                }
+            }
+
+            if (!policyData.isNull(resources.getString(R.string.policy_password_allow_simple)) &&
+                    policyData.get(resources.getString(R.string.policy_password_allow_simple)) != null) {
+
+                if (policyData.get(resources.getString(
+                        R.string.policy_password_allow_simple)) instanceof String) {
+                    complex =
+                            (String) policyData.get(resources.getString(
+                                    R.string.policy_password_allow_simple));
+                    if (!complex.equals(resources.getString(R.string.shared_pref_default_status))) {
+                        devicePolicyManager.setPasswordQuality(demoDeviceAdmin,
+                                DevicePolicyManager.PASSWORD_QUALITY_COMPLEX);
+                    }
+                } else if (policyData.get(resources.getString(
+                        R.string.policy_password_allow_simple)) instanceof Boolean) {
+                    isComplex =
+                            policyData.getBoolean(
+                                    resources.getString(R.string.policy_password_allow_simple));
+                    if (!isComplex) {
+                        devicePolicyManager.setPasswordQuality(demoDeviceAdmin,
+                                DevicePolicyManager.PASSWORD_QUALITY_COMPLEX);
+                    }
+                }
+            }
+
+            if (!policyData.isNull(resources.getString(R.string.policy_password_pin_age_in_days)) &&
+                    policyData.get(resources.getString(R.string.policy_password_pin_age_in_days)) !=
+                            null) {
+                int daysOfExp = policyData.getInt(resources.getString(R.string.policy_password_pin_age_in_days));
+                timout = daysOfExp * DAY_MILLISECONDS_MULTIPLIER;
+                devicePolicyManager.setPasswordExpirationTimeout(demoDeviceAdmin, timout);
+            }
+
+            if (Constants.DEBUG_MODE_ENABLED) {
+                Log.d(TAG, "Password policy set");
+            }
+        } catch (JSONException e) {
+            operation.setStatus(resources.getString(R.string.operation_value_error));
+            resultBuilder.build(operation);
+            throw new AndroidAgent
```

Comment:
Fixed in Commit [https://github.com/wso2/product-mdm/commit/b0c66506d24fb5288300bec4224f1eaa58c7b426]